packages/dataviews/src/components/dataform/stories/fixtures.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	ValidatedTextControl,
+	ValidatedTextareaControl,
+	ValidatedNumberControl,
+	ValidatedSelectControl,
+	ValidatedInputControl,
+} from '@automattic/components';
+
+/**
+ * Internal dependencies
+ */
+import type { Field, Form } from '../../../types';
+
+export type FormData = {
+	name: string;
+	surname: string;
+	email: string;
+};
+
+export const initialData: FormData = {
+	name: 'John',
+	surname: 'Doe',
+	email: 'john.doe@example.com',
+};
+
+export const getFields = (): Field< FormData >[] => [
+	{
+		id: 'name',
+		label: 'First Name',
+		type: 'text',
+		Edit: ( { field, data, onChange } ) => (
+			<ValidatedTextControl
+				label={ field.label }
+				value={ field.getValue( { item: data } ) }
+				onChange={ ( value: string ) =>
+					onChange( { [ field.id ]: value } )
+				}
+				customValidator={ () => {
+					return field?.isValid?.( data )
+						? undefined
+						: 'Name must be longer than 5 characters';
+				} }
+			/>
+		),
+		isValid: ( item ) => {
+			return item.name.length > 2;
+		},
+	},
+	{
+		id: 'surname',
+		label: 'Last Name',
+		type: 'text',
+		Edit: ( { field, data, onChange } ) => (
+			<ValidatedInputControl
+				label={ field.label }
+				value={ field.getValue( { item: data } ) }
+				onChange={ ( value ) => onChange( { [ field.id ]: value } ) }
+				customValidator={ () => {
+					const isValid = field?.isValid?.( data );
+					if ( typeof isValid === 'string' ) {
+						return isValid;
+					}
+
+					return undefined;
+				} }
+			/>
+		),
+		isValid: {
+			isRequired: true,
```

Comment:
Can we make it so this also passes the `required` prop to the `ValidatedInputControl`? That marks it as required in the HTML, and adds a visual indicator to the label.

Reply:
This is an interesting issue. Is it something that we should do or should be the responsibility of the consumer?

Reply:
No strong opinion on whether it should be part of the abstraction or not, but it probably should be clearly documented if DataForm isn't going to handle it. Any standard form element that's required should be marked as such in the HTML.

Reply:
Addressed with 9203476dc7cbcdbac88789975da569b8e0dff87a